### PR TITLE
[SchemaRegistry] update API for consistency

### PR DESCRIPTION
**Pull request by swathipil**
fixes: #20307

### CODE REVIEW

File: sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py [hunk 1]
```diff
@@ -120,29 +120,30 @@ def _get_schema(self, schema_id, **kwargs):
...
         :param schema: The schema used to encode the data.
         :type schema: Union[str, bytes]
-        :return:
+        :rtype: bytes
         """
         raw_input_schema = schema
         try:
             cached_schema = self._user_input_schema_cache[raw_input_schema]
-        except KeyError:
+        except (KeyError, TypeError):
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py:139]: in what cases will the `TypeError` be raised for input schema type of string/bytes?

File: sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py [hunk 2]
```diff
@@ -120,29 +120,30 @@ def _get_schema(self, schema_id, **kwargs):
...
         # type: (Dict[str, Any], Union[str, bytes], Any) -> bytes
         """
-        Encode dict data with the given schema. The returns bytes are consisted of: The first 4 bytes
+        Encode data with the given schema. The returns bytes are consisted of: The first 4 bytes
         denoting record format identifier. The following 32 bytes denoting schema id returned by schema registry
         service. The remaining bytes are the real data payload.
 
-        :param data: The dict data to be encoded.
+        :param value: The data to be encoded.
+        :type value: Dict[str, Any]
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py:131]: as per OOB discussion, we need to investigate the input of Any type (python primitive types)

File: sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py [hunk 3]
```diff
@@ -155,21 +156,21 @@ def serialize(self, data, schema, **kwargs):
...
 
-    def deserialize(self, data, **kwargs):
+    def deserialize(self, value, **kwargs):
         # type: (bytes, Any) -> Dict[str, Any]
         """
         Decode bytes data.
 
-        :param bytes data: The bytes data needs to be decoded.
+        :param bytes value: The bytes data needs to be decoded.
         :rtype: Dict[str, Any]
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py:165]: as per OOB discussion, we need to investigate supporting input of Any type (python primitive types) first, which could change this API to return Any type

File: sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py (hunk 1 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py:139]: ah sorry, I forgot to remove this when I was fixing a bug

**Review by yunhaoling — APPROVED** [review]
LGTM! thanks!

Also we need to update the changelog for the breaking changes, but we could do it in a separate doc update PR.

**Pull request merged**